app/src/main/java/com/example/xpvehicles/activities/VehicleDetailsActivity.java

```diff
@@ -87,26 +109,23 @@ private void bind() {
     }
 
     private void setValues() {
-        String distanceFromUser = vehicle.getDistanceFromUser();
+        if (MainActivity.getUserLocationGeoPoint() != null) {
+            int distanceFromUser = MainActivity.getDistanceFromUser(vehicle);
+            tvDetailsDistanceFromUser.setText(distanceFromUser + "mi.");
+        }
```

Comment:
I see why now it is static. We shouldn't do this. 
Instead, we should use intent to pass the data.

Reply:
Should I pass the distance in the intent? Since getting the distance is asynchronous how can I ensure that I have the distance before a user clicks on a vehicle to see the detail?

Reply:
When I pass the data, I run into the potential issue when a user can click on a vehicle before the distance has been calculated, and it displays a default value I set to 0.